### ISSUE TITLE
Cursor disappearing

### ISSUE THREAD

**Issue by NickStafford2** [lukas-reineke/indent-blankline.nvim#932]
### Problem

The cursor occasionally disappears behind the vertical lines. I am still trying to figure out why it occurs. I thought I would report it for the record. 

I am investigating it myself and I will post a fix/explanation if I figure it out. 

## Description
A difficult bug to reproduce, but I have seen it several times. If I can replicate it again, I will add further notes. 

I have no configuration settings for indent-blankline. I am using Lazy-vim with some custom plugins. I can provide details if needed. If anyone can point me in the right direction, I am happy to help write up a fix. 

## indent-breakline version inside lazy-lock.json
`lazy-lock.json:25:4:  "indent-blankline.nvim": { "branch": "master", "commit": "3af6493bf69e4a857a8b1fab36f333629d413a18" }, 
`
## Nvim
- NVIM v0.10.1
- Build type: Release
- LuaJIT 2.1.1713484068

### Expected behavior

Cursor remains visible.

### Neovim version (nvim -v)

v0.10.1

**Comment by lukas-reineke** [comment]
IBL does not do anything with the cursor. This looks like a lower level bug. Either with Neovim or your terminal.  
I don't think I can help you with this, sorry.

**Comment by NickStafford2** [comment]
Thanks. I will keep investigating deeper down. If I fix it, I could post back here in case it helps anyone else. 

Disabling this plugin does fix it, but I'll keep debugging.

**Comment by nacht-falter** [comment]
I had the same issue and can confirm that disabling the plugin fixes this.

This is related to the cursor color. Here are a few ideas: Try disabling the cursorline (`:set nocul`) or change the cursorline color (e.g. `:highlight CursorLine guibg=#1e1e1e`) and see if that fixes it. You could also set a cursor color in your terminal settings, which should override the neovim setting.

Hope that helps!

**Comment by Hummmmmmmmus** [comment]
I've been having the same issue and I think it may be related to [this](https://github.com/lukas-reineke/indent-blankline.nvim/blob/e10626f7fcd51ccd56d7ffc00883ba7e0aa28f78/lua/ibl/init.lua#L484) line, specifically the fact that when the virtual text for the indent hints is given to the buffer it is set to the "combine" highlight mode. The problem with this, I *think*, is that combine mode changes the color scheme of the cursor in the process of combining it with the virtual text (in this case the indent hint character), however this color change makes it blend in with the cursor line (when its enabled) so it *looks* as if the cursor disappeared. Barring any weirdness with the virtual text I don't know about, the solution to this may be as simple as adding an extra config.indent setting for the hl_mode in that call.

**Comment by lukas-reineke** [comment]
No, that is not how `hl_mode` works. It only affects the virtual text. Please read the documentation `:help nvim_buf_set_extmark()`

**Comment by Hummmmmmmmus** [comment]
Yeah, that was waaay off, got a little carried away with something from the docs I didn't read thoroughly. 

If not that then it seems like this is exclusively a cursor/indent guide color scheme issue resulting in the cursor blending into the cursorline when its hovering over the indent guides. I think specifically its because the highlight of the indent guides gets its coloring from the whitespace group which just so happens to be the same color as the cursorline group (not sure if thats common or not but it is in my color scheme so ¯\_(ツ)_/¯). Seems like a working fix is just to change the highlight group of the indent characters away from Whitespace in the config. Another fix without changing the colorscheme would maybe be to make the cursor completely replace the virtual text its hovering over rather than trying to merge with the virtual text but as you could probably tell from my first message, I don't know how to do that.

**Comment by LorenzBischof** [comment]
The following snippet removes the virtual text character underneath the cursor. I have no idea of the performance implications 😆

```lua
local ns_id = vim.api.nvim_create_namespace('indent_blankline')
vim.api.nvim_create_augroup("CursorHighlightGroup", { clear = true })
vim.api.nvim_create_autocmd("CursorMoved", {
  group = "CursorHighlightGroup",
  callback = function()
    local line, col = unpack(vim.api.nvim_win_get_cursor(0))
    local char = vim.api.nvim_get_current_line():sub(col + 1, col + 1)
    if char == " " then
      local bufnr = vim.api.nvim_get_current_buf()
      vim.api.nvim_buf_set_extmark(bufnr, ns_id, line - 1, col, {
        virt_text = { { " ", "NonText" } },
        virt_text_pos = 'overlay',
        hl_mode = 'combine'
      })
    end
  end,
})
```
